File: backend/src/main/java/iuh/fit/backend/model/CartItem.java
package iuh.fit.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

// Cart Item Entity
@Entity
@Table(name = "cart_items")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class CartItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "cart_id")
    private Integer cartId;

    @Column(name = "product_variant_id")
    private Integer productVariantId;

    private Integer quantity;

    @ManyToOne
    @JoinColumn(name = "cart_id", insertable = false, updatable = false)
    @JsonIgnore
    private Cart cart;

    @OneToOne
    @JoinColumn(name = "product_variant_id", insertable = false, updatable = false)
    @JsonIgnore
    private ProductVariant productVariant;
}

File: frontend/src/types/CartItem.ts
import type { Cart } from './Cart';
import type { ProductVariant } from './ProductVariant';

export interface CartItem {
  id: number;
  cartId?: number;
  productVariantId: number;
  productName: string;
  variantName: string;
  quantity: number;
  price: number;
  subtotal: number;
  imageUrl?: string;
  cart?: Cart;
  productVariant?: ProductVariant;
}
